src/wire/icmpv4.rs

```diff
@@ -384,7 +385,7 @@ pub enum Repr<'a> {
 impl<'a> Repr<'a> {
     /// Parse an Internet Control Message Protocol version 4 packet and return
     /// a high-level representation.
-    pub fn parse<T: AsRef<[u8]> + ?Sized>(packet: &Packet<&'a T>) -> Result<Repr<'a>> {
+    pub fn parse<T: AsRef<[u8]> + ?Sized>(packet: &Packet<&'a T>, _checksum: &ChecksumCapabilities) -> Result<Repr<'a>> {
```

Comment:
Shouldn't we verify checksum in this method? Looks like I forgot to implement it.

Reply:
Yeah I'm not sure what the story is around there, that's what I meant with:
`ICMPv4 checksums don't seem to be implemented? (unused checksum warning in icmpv4::parse)`
in my first comment.

I'll include that in the next round:
```rust
if checksum_caps.icmpv4.rx() && !packet.verify_checksum() { 
            return Err(Error::Checksum) 
        }
```